fabric-ingredient-api-v1/src/main/java/net/fabricmc/fabric/api/ingredient/v1/CustomIngredient.java

```diff
@@ -0,0 +1,20 @@
+package net.fabricmc.fabric.api.ingredient.v1;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.recipe.Ingredient;
+
+import net.fabricmc.fabric.impl.ingredient.CustomIngredientImpl;
+
+// TODO: what about getMatchingItemIds ?
+public interface CustomIngredient {
+	boolean matchesStack(ItemStack stack);
```

Comment:
An Ingredient is basically a `Predicate<ItemStack>` so why not have this as
```suggestion
	boolean test(ItemStack stack);
```